editor/src/components/shared/scoped-variables.ts

```diff
@@ -83,26 +99,79 @@ function getVariablesFromComponent(
   const jsxComponentVariables = variablesInScopeFromEditorState[elementPathString] ?? {}
   return {
     filePath: jsxComponent?.name ?? 'Component',
-    variables: Object.entries(jsxComponentVariables).flatMap(([name, value]) => {
-      const type = getTypeByValue(value)
-      const variable = {
-        name,
-        value,
-        type,
+    variables: generateVariableTypes(jsxComponentVariables),
+  }
+}
+
+function getComponentPropsInScope(
+  topLevelElements: TopLevelElement[],
+  elementPath: ElementPath,
+  jsxMetadata: ElementInstanceMetadataMap,
+): AllVariablesInScope {
+  const jsxComponent = topLevelElements.find((el) => isUtopiaJSXComponent(el)) as UtopiaJSXComponent
+  const jsxComponentPropNamesUsed = jsxComponent?.propsUsed ?? []
+
+  const jsxComponentProps = omitWithPredicate(
+    getContainerPropsValue(elementPath, jsxMetadata),
+    (key) => !jsxComponentPropNamesUsed.includes(key as string),
+  )
+
+  const name = jsxComponent?.name ?? 'Component'
+  return {
+    filePath: `${name}::props`,
+    variables: generateVariableTypes(jsxComponentProps),
+  }
+}
+
+function getContainerPropsValue(
+  elementPath: ElementPath,
+  jsxMetadata: ElementInstanceMetadataMap,
+): VariableData {
+  const containerPathString = toComponentId(getContainingComponent(elementPath))
+  const containerMetadata = jsxMetadata[containerPathString]?.element.value
+  const runtimeProps: VariableData = {}
+
+  if (
+    containerMetadata != null &&
+    typeof containerMetadata !== 'string' &&
+    isJSXElement(containerMetadata)
+  ) {
+    containerMetadata.props.forEach((prop) => {
+      if (isJSXAttributesEntry(prop)) {
+        const value = simplifyAttributeIfPossible(prop.value)
+        if (!isJSXAttributeValue(value)) return
```

Comment:
Is it not simpler to just enclose the next line in the conditional but flipped?
```suggestion
        if (isJSXAttributeValue(value)) {
```

Reply:
Good catch, it was actually like that and it was a result of a bad refactor. Thanks!